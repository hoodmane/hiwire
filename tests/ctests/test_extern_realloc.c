#include "testlib.h"

HwRef blah[10000];

void*
hiwire_realloc(void* orig, size_t sz)
{
  printf("realloc! %d\n", sz);
  if (sz > sizeof(blah)) {
    return 0;
  }
  return blah;
}

int
main()
{
  int offset = 80000;
  HwRef refs[5000];
  for (int i = 0; i < 5000; i++) {
    refs[i] = hiwire_new(int_to_ref(offset + i));
  }
  for (int i = 0; i < 5000; i++) {
    if (ref_to_int(hiwire_get(refs[i])) != offset + i) {
      printf(FAIL_STR " %d\n", i);
      return 1;
    }
  }
  printf(PASS_STR "\n");
  for (int i = 0; i < 5000; i++) {
    hiwire_decref(refs[i]);
  }
  offset *= 6;
  for (int i = 0; i < 5000; i++) {
    refs[i] = hiwire_new(int_to_ref(offset + 2 * i));
  }
  for (int i = 0; i < 5000; i++) {
    if (ref_to_int(hiwire_get(refs[i])) != offset + 2 * i) {
      printf(FAIL_STR "\n");
      return 1;
    }
  }
  printf(PASS_STR "\n");

  return 0;
}
