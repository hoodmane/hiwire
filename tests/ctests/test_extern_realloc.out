realloc! 4096
realloc! 8192
realloc! 12288
realloc! 16384
realloc! 20480
[94;1mPASS [0m
[94;1mPASS [0m
